designs/per_command_help_text.md

```diff
@@ -0,0 +1,386 @@
+                                  Key=InstanceType,ParameterValue=t1.micro' or
+                                  KeyPairName=MyKey InstanceType=t1.micro
+  --debug                         Turn on debug logging to print debug message
+                                  generated by AWS SAM CLI and display
+                                  timestamps.
+  --profile TEXT                  Select a specific profile from your
+                                  credential file to get AWS credentials.
+  --region TEXT                   Set the AWS Region of the service (e.g. us-
+                                  east-1).
+  --metadata                      Optional. A map of metadata to attach to ALL
+                                  the artifacts that are referenced in your
+                                  template.
+  --notification-arns LIST        Amazon  Simple  Notification  Service
+                                  topicAmazon  Resource  Names  (ARNs) that
+                                  AWS CloudFormation associates withthe stack.
+  --tags                          A list of tags to associate with the stack
+                                  that is created or updated.AWS
+                                  CloudFormation also propagates these tags to
+                                  resources in the stack if the resource
+                                  supports it.
+  --capabilities LIST             A list of capabilities that you must specify
+                                  before AWS Cloudformation can create certain
+                                  stacks. Some stack templates might include
+                                  resources that can affect permissions in
+                                  your AWS account, for example, by creating
+                                  new AWS Identity and Access Management (IAM)
+                                  users. For those stacks, you must explicitly
+                                  acknowledge their capabilities by specifying
+                                  this parameter. The only valid valuesare
+                                  CAPABILITY_IAM and CAPABILITY_NAMED_IAM. If
+                                  you have IAM resources, you can specify
+                                  either capability. If you have IAM resources
+                                  with custom names, you must specify
+                                  CAPABILITY_NAMED_IAM. If you don't specify
+                                  this parameter, this action returns an
+                                  InsufficientCapabilities error.
+  -h, --help                      Show this message and exit.
+
+```
+
+Users would probably have a ton of questions based on that help text. Some things that come up top of mind are below.
+
+* Which options are required?
+* What are the options that are AWS infra related? AWS Credentials related?
```

Comment:
I really like this. I think we should go one step further and specify if a command itself requires AWS credentials or not. One of the first things in the help text should be something like:
```
AWS credentials ARE required ~ this command interacts with AWS APIs, meaning that valid AWS credentials are required for this command to be run.
```
or
```
AWS credentials ARE NOT required ~ this command does not interact with AWS APIs, and makes changes that are localized to executing machine.
```